src/utils/bioschemasMapper.js

```diff
@@ -0,0 +1,196 @@
+        name: 'hermaphrodite'
+      },
+      NAV: {
+        label: 'Not Available'
+      },
+      NASK: {
+        label: 'Not Asked'
+      },
+      UNKNOWN: {
+        label: 'unknown'
+      }
+    }
+  }
+  return property in mappings ? mappings[property][term] : undefined
+}
+
+const getCollectionAdditionalProperty = (data, propertyName) => {
+  let value
+
+  if ('uri' in data) {
+    // it means the data contains the uri of the code in the model and must be used
+    value = {
+      '@type': 'CategoryCode',
+      '@id': data.uri,
+      codeValue: data.uri.split('/').slice(-1).pop() || undefined
+    }
+  } else {
+    // Split values such as /api/v2/eu_bbmri_eric_material_types/TISSUE_FROZEN
+    const codeValue = 'id' in data ? data.id : data._href.split('/').slice(-1).pop()
+    const ontologyTerm = getOntologyTerm(propertyName, codeValue)
+    // it gets mappings from the static object. It should be better to include the uri in the model
+    if (ontologyTerm) {
+      if ('uri' in ontologyTerm) {
+        value = {
+          '@type': 'CategoryCode',
+          '@id': ontologyTerm.uri,
+          codeValue: ontologyTerm.code
+        }
+      } else {
+        value = ontologyTerm.label
+      }
+    } else { // if no ontology term is found, it adds the label of the value
+      value = data.label || data.size
+    }
+  }
+  return {
+    '@type': 'PropertyValue',
+    propertyId: getOntologyTerm('properties', propertyName) || undefined,
+    name: propertyName,
+    value: value
+  }
+}
+
+export const mapCollectionsData = (collection) => {
+  const jsonld = {
+    '@context': 'https://schema.org',
+    '@type': 'Dataset',
+    '@id': `${getBaseUrl()}/collection/${collection.id}`,
+    description: collection.description,
+    identifier: collection.id,
+    keywords: 'sample, collection',
+    name: collection.name,
+    url: `${getBaseUrl()}/collection/${collection.id}`,
+    includedInDataCatalog: {
+      '@type': 'DataCatalog',
+      '@id': `${getBaseUrl()}/biobank/${collection.biobank.id}`,
+      name: collection.biobank.name,
+      url: `${getBaseUrl()}/biobank/${collection.biobank.id}`
+    },
+    additionalProperty: []
+  }
+  const properties = ['diagnosis_available', 'materials', 'data_categories', 'storage_temperatures', 'sex', 'order_of_magnitude']
+  properties.forEach(prop => {
+    if (prop in collection) {
+      if (collection[prop] instanceof Array) {
+        collection[prop].forEach(item =>
+          jsonld.additionalProperty.push(getCollectionAdditionalProperty(item, prop))
+        )
+      } else {
+        jsonld.additionalProperty.push(getCollectionAdditionalProperty(collection[prop], prop))
+      }
+    }
+  })
+  return jsonld
+}
+
+export const mapBiobankDataCatalog = (biobank) => {
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'DataCatalog',
+    '@id': `${getBaseUrl()}/biobank/${biobank.id}`, // TODO: Change with the persistent identifier
```

Comment:
maybe address this todo?

Reply:
Currently, to my knowledge, the biobanks don't have a persistent universal identifier, but I understood that there is a task about this. This is a reminder to change this whenever the persistent uid will be available